rpm/0045-sailfishos-gecko-Fix-search-directories.-JB-55267.patch

```diff
@@ -0,0 +1,58 @@
+From 9672c13462240646e580d2a81868b54af1984fce Mon Sep 17 00:00:00 2001
+From: Marko Kenttala <marko.kenttala@jolla.com>
+Date: Fri, 10 Sep 2021 13:51:56 +0300
+Subject: [PATCH] [sailfishos][gecko] Fix search directories. JB#55267
+
+---
+ xpcom/io/nsAppFileLocationProvider.cpp | 21 ++++++++++++++++++++-
+ 1 file changed, 20 insertions(+), 1 deletion(-)
+
+diff --git a/xpcom/io/nsAppFileLocationProvider.cpp b/xpcom/io/nsAppFileLocationProvider.cpp
+index 915e8bc7361e..2f60a768fd42 100644
+--- a/xpcom/io/nsAppFileLocationProvider.cpp
++++ b/xpcom/io/nsAppFileLocationProvider.cpp
+@@ -58,6 +58,9 @@
+ #define CHROME_DIR_NAME NS_LITERAL_CSTRING("chrome")
+ #define PLUGINS_DIR_NAME NS_LITERAL_CSTRING("plugins")
+ 
++#define NS_SYSTEM_SEARCH_DIR "SystemSearch"
++#define NS_USER_SEARCH_DIR "UserSearch"
++
+ //*****************************************************************************
+ // nsAppFileLocationProvider::Constructor/Destructor
+ //*****************************************************************************
+@@ -181,6 +184,19 @@ nsAppFileLocationProvider::GetFile(const char* aProp, bool* aPersistent,
```

Comment:
I'd expect DirProvider::GetFile getting called (embedding/embedlite/utils/DirProvider.cpp) as GeckoLoader initializes DirProvider and passes handle down to NS_InitXPCOM where it is getting registered to the nsDirectoryService. So, rather than patching we could have an implementation there in the DirProvider.cpp.

```
GeckoLoader::InitEmbedding
XRE_InitEmbedding2
NS_InitXPCOM (XPCOMInit.cpp)
nsDirectoryService::gService->RegisterProvider(aAppFileLocationProvider);
```

Reply:
Yep, seems to work. Copied the change there.